homeassistant/components/homekit_controller/select.py

```diff
@@ -0,0 +1,69 @@
+"""Support for Homekit select entities."""
+from __future__ import annotations
+
+from aiohomekit.model.characteristics import Characteristic, CharacteristicsTypes
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import KNOWN_DEVICES, CharacteristicEntity
+
+_ECOBEE_MODE_TO_TEXT = {
+    0: "Home",
+    1: "Sleep",
+    2: "Away",
+}
+_ECOBEE_MODE_TO_NUMBERS = {v: k for (k, v) in _ECOBEE_MODE_TO_TEXT.items()}
+
+
+class EcobeeModeSelect(CharacteristicEntity, SelectEntity):
+    """Represents a ecobee mode select entity."""
+
+    _attr_options = ["Home", "Sleep", "Away"]
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device if any."""
+        if name := super().name:
+            return f"{name} Current Mode"
+        return "Current Mode"
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity cares about."""
+        return [
+            CharacteristicsTypes.Vendor.ECOBEE_CURRENT_MODE,
+        ]
+
+    @property
+    def current_option(self) -> str:
```

Comment:
You could also give this a custom device class if you wanted to make the strings translatable.  Example in wled

Reply:
Seems like I just need to make one up and prefix it with DOMAIN and `__`? Is that really all?

Reply:
If it is, then done.

Reply:
You'll need a `strings.select.json` as well

Reply:
Rebased to fix conflict and hopefully wired up translations.